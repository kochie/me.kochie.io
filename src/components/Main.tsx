import * as React from "react";
import { isMobile } from "react-device-detect";

import Header from "./Header";
import Footer from "./Footer";
import Middle from "./Middle";

import "../styles/Main.scss";

export default class Main extends React.Component<{}, {}> {
  render() {
    return (
      <div className="main">
        {/* {isMobile || (window.innerWidth <= 1920 && window.devicePixelRatio > 2) ? null : <Header />} */}
        <Middle />
        <Footer />
      </div>
    );
  }
}
