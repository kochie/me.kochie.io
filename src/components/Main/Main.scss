.main {
  grid-area: main;
  min-height: 100vh;
  display: grid;
  grid-template-rows: 1fr auto;
  grid-template-columns: 1fr;
  grid-template-areas: "middle" "footer";
}
