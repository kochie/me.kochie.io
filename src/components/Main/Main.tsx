import * as React from "react";

import { Footer } from "..";
import { Middle } from "..";
import Head from "next/head";

export default class Main extends React.Component<{}, {}> {
  render() {
    return (
      <>
        <Head>
          <link rel="stylesheet" href="/static/styles/main.css" />
        </Head>
        <div className="main">
          <Middle />
          <Footer />
        </div>
      </>
    );
  }
}
