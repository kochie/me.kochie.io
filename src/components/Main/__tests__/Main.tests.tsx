import * as React from "react";
import * as renderer from "react-test-renderer";

import Main from "../Main";

it("main renders correctly", () => {
  const tree = renderer.create(<Main />).toJSON();
  expect(tree).toMatchSnapshot();
});
