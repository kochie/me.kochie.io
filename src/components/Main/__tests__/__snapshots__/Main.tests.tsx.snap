// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main renders correctly 1`] = `
<div
  className="main"
>
  <div
    className="middle"
  >
    <h1>
      I’m Rob, a university student from Melbourne, Australia.
    </h1>
    <p>
      I currently study Electrical Engineering, Computer Science, and Mathematics at Monash University where I was the president of the IT society 
      <a
        href="http://wired.org.au"
      >
        wired
      </a>
       in 2017.
    </p>
    <p>
      I'm also apart of the 
      <a
        href="https://unihack.net"
      >
        Unihack
      </a>
       team based in Melbourne, we run hackathons for students around Australia. 
    </p>
    <p>
      Apart from that, I love gaming, karate, bike riding, and electronics and try to do at least one of these things every day. I’m always looking for something new and interesting to do, so feel free to browse my portfolio or view my resume if you want to see some of the cool things I’ve done.
    </p>
    <div
      className="bmcButtonContainer"
    >
      <a
        className="bmcButton"
        href="https://www.buymeacoffee.com/kochie"
        target="_blank"
      >
        <img
          alt="Buy me a coffee"
          src="https://www.buymeacoffee.com/assets/img/BMC-btn-logo.svg"
        />
        <span
          className="bmcSpan"
        >
          Buy me a coffee
        </span>
      </a>
    </div>
    <hr />
    <div>
      <h2>
        Recent Work
      </h2>
      <div
        className="portfolio"
      >
        <div
          className="portfolioItem"
        >
          <figure
            className="portfolioImageWrapper"
          >
            <a
              href="https://kochie.space"
            >
              <img
                className="portfolioImage"
                src="/images/space.png"
              />
            </a>
          </figure>
          <h3>
            Space
          </h3>
          <p>
            A small webapp that follows the mouse.
          </p>
        </div>
        <div
          className="portfolioItem"
        >
          <figure
            className="portfolioImageWrapper"
          >
            <a
              href="https://3body.now.sh"
            >
              <img
                className="portfolioImage"
                src="/images/particlez.png"
              />
            </a>
          </figure>
          <h3>
            Particlez
          </h3>
          <p>
            A real time n-body particle simulator for the web.
          </p>
        </div>
        <div
          className="portfolioItem"
        >
          <figure
            className="portfolioImageWrapper"
          >
            <a
              href="https://github.com/kochie/guardian-server"
            >
              <img
                className="portfolioImage"
                src="/images/guardian.png"
              />
            </a>
          </figure>
          <h3>
            Guardian
          </h3>
          <p>
            An experiment in transferring cyptographic keys between devices without the user needing to copy keys themselves.
          </p>
        </div>
        <div
          className="portfolioItem"
        >
          <figure
            className="portfolioImageWrapper"
          >
            <a
              href="https://bitbucket.org/kochie/fyp"
            >
              <img
                className="portfolioImage"
                src="/images/tensorflow.jpeg"
              />
            </a>
          </figure>
          <h3>
            Optimising Neural Networks for Embedded Systems
          </h3>
          <p>
            My final year project - focusing on modifying the activation layers of convolutional neural networks to work better on embedded devices.
          </p>
        </div>
        <div
          className="portfolioItem"
        >
          <figure
            className="portfolioImageWrapper"
          >
            <a
              href="https://github.com/kochie/juice"
            >
              <img
                className="portfolioImage"
                src="/images/juice.png"
              />
            </a>
          </figure>
          <h3>
            Juice
          </h3>
          <p>
            A Json-Web-Token analyser.
          </p>
        </div>
        <div
          className="portfolioItem"
        >
          <figure
            className="portfolioImageWrapper"
          >
            <a
              href="https://github.com/kochie/mcgurrin"
            >
              <img
                className="portfolioImage"
                src="/images/keyboard.png"
              />
            </a>
          </figure>
          <h3>
            McGurrin
          </h3>
          <p>
            A tool to help with learning to touch type.
          </p>
        </div>
      </div>
    </div>
    <hr />
    <div>
      <h2>
        Recent Articles
      </h2>
      <div
        className="card"
      >
        <div>
          <a
            href="https://medium.com/unihack-blog/why-build-an-app-when-you-can-build-a-phone-88b5ce7d0104"
          >
            <img
              src="https://cdn-images-1.medium.com/max/800/1*ijymAR61qIf690CHOV3Hcw.jpeg"
            />
          </a>
        </div>
        <div>
          <h3>
            Why build an app when you can build a phone?
          </h3>
          <p>
            Over the last 5 years, UNIHACK has been predominantly focused on software and design. It is not well known that some of UNIHACK’s most creative entire
            ... 
            <a
              href="https://medium.com/unihack-blog/why-build-an-app-when-you-can-build-a-phone-88b5ce7d0104"
            >
              more
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      className="button"
    />
  </div>
  <nav
    className="footer"
  >
    <ul>
      <li>
        Robert Koch
      </li>
      <li>
        <span>
          Updated on 
          <time />
        </span>
      </li>
    </ul>
  </nav>
</div>
`;
